src/ArchiveUtils.jl

```diff
@@ -146,3 +148,37 @@ function download_verify(url, hash, path)
         verify(path, hash) || error("Verification failed")
     end
 end
+
+# Copy of `Pkg.Artifacts.archive_artifact` that supports a custom `package` function
+function _archive_artifact(hash::SHA1, tarball_path::String;
+                           honor_overrides::Bool=false,
+                           package::Function=_package_fast)
+
+    if !honor_overrides
```

Comment:
You can drop `honor_overrides`

Reply:
Sure, I just kept the function as close to the original as possible